### ISSUE TITLE
[Enhancement] iOS OnPlatform Secondary ToolbarItem Overflow

### ISSUE THREAD

**Issue by ChaseFlorell** [xamarin/Xamarin.Forms#7557]
## Summary
In an effort to make our UI's consistent across platforms, I'd like to propose an iOS specific platform property to change the secondary ToolbarItem.

## API Changes

```csharp
// new
public enum ToolbarStyle
{
    Default,
    Overflow
}
```

```xaml
<Page xmlns:ios="clr-namespace:MyCompany.Forms.PlatformConfiguration.iOS"
      ios:Toolbar.SecondaryToolbarStyle="Overflow"
      ios:Toolbar.OverflowImageSource="vertical_ellipsis.png">

</Page>
```

## Intended Use Case
Provide the ability to show Secondary Toolbar Items as Overflow on iOS akin to Android
![overflow_menu_2](https://user-images.githubusercontent.com/471626/65077607-21d92200-d950-11e9-8da7-519f29ff7f11.png)

**Comment by adammeaney** [comment]
I was considering doing this, wanted to check in if we still thought it was reasonable to do by page, instead of by NavigationPage or entire application.

I can't see a great reason for this not to for a whole navigation page, plus it makes the code of this much simpler as I don't have to come up with a good way to communicate from the pagerenderer to the navigationpage renderer to get the change, remove the toolbar, etc.

I just would assume that if you changed the style, it would look very weird to have it mixed, where some pages are default and some are overflow.

**Comment by ChaseFlorell** [comment]
I have a working implementation of this using Platform Specifics and an Effect, however I'm not sure how to get the UITableView to partially overlap the Toolbar.

**Comment by ChaseFlorell** [comment]
I have successfully built this and would be happy to submit a PR if the Forms team is willing to accept it. The API I used might be a little jankey, so I'm happy to get some feedback on how to improve it.

The idea is that when the attached property is set on a NavigationPage, we start listening to Pushed/Popped events. Each page that is pushed gets an Effect attached, and it's removed when popped. The effect attaches a UITableView to the RootView in iOS (with a little offset).

Note, I have not done anything with Shell, so I'm not sure if the same applies ¯\_(ツ)_/¯

<img width="457" alt="Screen Shot 2019-11-19 at 6 43 32 AM" src="https://user-images.githubusercontent.com/471626/69156209-ed8cf980-0a97-11ea-894d-f1bc6aa5f1db.png">
